business/apps.go

```diff
@@ -192,3 +195,31 @@ func fetchNamespaceApps(k8s kubernetes.IstioClientInterface, namespace string, a
 
 	return castAppDetails(services, ws), nil
 }
+
+func (in *AppService) fillCustomDashboardRefs(namespace string, app *models.App, details *appDetails) {
+	app.CustomDashboards = []models.CustomDashboardRef{}
+	uniqueRefs := make(map[string]string)
+	for _, workload := range details.Workloads {
+		for _, pod := range workload.Pods {
+			for _, ref := range pod.CustomDashboards {
+				if ref != "" {
+					uniqueRefs[ref] = ref
+				}
+			}
+		}
+	}
+	mon, err := kubernetes.NewKialiMonitoringClient()
+	if err != nil {
+		log.Error("Cannot get dashboards - cannot load Kiali Monitoring Client")
+		return
+	}
+	for _, ref := range uniqueRefs {
+		title := getCustomDashboardTitle(mon, namespace, ref)
```

Comment:
Ok, I understand that the Pods have the annotation if there is some custom dashboard available.
Then, with those ref, we fetch just the title or the whole dashboard definition ?

I miss some comments with the intention of the logic here as well.

Reply:
Looks like I messed up with my commits, this one was about KIALI-1975... it shouldn't be here. I'll push again, sorry :-(
(which doesn't invalidate your remark, I can add a code comment)